internal/js_library/js_library.bzl

```diff
@@ -46,23 +48,66 @@ def write_amd_names_shim(actions, amd_names_shim, targets):
                 amd_names_shim_content += "define(\"%s\", function() { return %s });\n" % n
     actions.write(amd_names_shim, amd_names_shim_content)
 
-def _js_library(ctx):
-    return [
+def _impl(ctx):
+    if not ctx.files.srcs:
+        fail("No srcs specified")
```

Comment:
could it just be `mandatory = True` ?

Reply:
not quite as you can still specify `srcs = []` which satisfies mandatory. but I can simplify with
```
is_all_sources = ctx.files.srcs and ctx.files.srcs[0].is_source
```
instead